internal/authz/authz_test.go

```diff
@@ -179,3 +311,198 @@ func TestIndividualAuthorize(t *testing.T) {
 		require.NoError(t, err)
 	})
 }
+
+func TestModuleAuthorize(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	mockServer := mocks.NewMockServerInterface(mockController)
+
+	authorizer := NewAuthorizer(&Config{StoreID: "test-store", ModelID: "test-model"}, mockServer, logger.NewNoopLogger())
+
+	t.Run("return_no_error_when_no_modules", func(t *testing.T) {
+		err := authorizer.moduleAuthorize(context.Background(), "client-id", CanCallWrite, "store-id", []string{})
+		require.NoError(t, err)
+	})
+
+	t.Run("error_when_first_module_errors", func(t *testing.T) {
+		mockServer.EXPECT().Check(gomock.Any(), gomock.Any()).Return(&openfgav1.CheckResponse{Allowed: false}, nil)
+		mockServer.EXPECT().Check(gomock.Any(), gomock.Any()).Return(&openfgav1.CheckResponse{Allowed: true}, nil)
+		mockServer.EXPECT().Check(gomock.Any(), gomock.Any()).Return(&openfgav1.CheckResponse{Allowed: true}, nil)
+
+		err := authorizer.moduleAuthorize(context.Background(), "client-id", CanCallWrite, "store-id", []string{"module1", "module2", "module3"})
+		require.Error(t, err)
+		require.Equal(t, "rpc error: code = Code(403) desc = the principal is not authorized to perform the action", err.Error())
```

Comment:
nitpik (non-blocking): You don't need to do two assert. You can just call

require.Equal(t, status.Error(codes.Code..... ), err).